storage/cloud-client/encryption.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This application demonstrates how to upload and download encrypted blobs
+(objects) in Google Cloud Storage.
+
+Use `generate-encryption-key` to generate an example key:
+
+    python encryption.py generate-encryption-key
+
+Then use the key to upload and download files encrypted with a custom key.
+
+For more information, see the README.md under /storage and the documentation
+at https://cloud.google.com/storage/docs/encryption.
+"""
+
+import argparse
+import base64
+import os
+
+from gcloud import storage
+
+
+def generate_encryption_key():
+    """Generates a 256 bit (32 byte) AES encryption key and prints the
+    base64 representation.
+
+    This is included for demonstration purposes. You should generate your own
+    key. Please remember that encryption keys should be handled with a
+    comprehensive security policy.
+    """
+    key = os.urandom(32)
+    encoded_key = base64.b64encode(key).decode('utf-8')
+    print('Base 64 encoded encryption key: {}'.format(encoded_key))
```

Comment:
What do you think about returning the value, rather than printing it?
I'm of two minds about it:
* I think in a real-world scenario / reading this script directly, it makes more sense to return the value and have the calling function do the printing
* As a code snippet, though, it's cleaner to just print the value - ie if you're just including this method as a snippet in the docs. In this case, however, I usually name the function with a `print_` prefix to make this behavior clear